graylog2-web-interface/src/components/metrics/MetricDetails.jsx

```diff
@@ -14,67 +14,106 @@
+    > dt {
+      float: left;
+    }
 
-const MetricDetails = createReactClass({
-  displayName: 'MetricDetails',
-
-  propTypes: {
-    metric: PropTypes.object.isRequired,
-  },
+    &.metric-timer > dd {
+      margin-left: 145px;
+    }
 
-  mixins: [Reflux.connect(MetricsStore)],
+    &.metric-meter > dd {
+      margin-left: 115px;
+    }
 
-  componentDidMount() {
-    MetricsActions.add(this.props.nodeId, this.props.metric.full_name);
-  },
+    &.metric-gauge > dd {
+      margin-left: 90px;
+    }
 
-  componentWillUnmount() {
-    MetricsActions.remove(this.props.nodeId, this.props.metric.full_name);
-  },
+    &.metric-counter > dd {
+      margin-left: 90px;
+    }
 
-  _formatDetailsForType(type, metric) {
-    switch (type) {
-      case 'Counter':
-        return <CounterDetails metric={metric} />;
-      case 'Gauge':
-        return <GaugeDetails metric={metric} />;
-      case 'Histogram':
-        return <HistogramDetails metric={metric} />;
-      case 'Meter':
-        return <MeterDetails metric={metric} />;
-      case 'Timer':
-        return <TimerDetails metric={metric} />;
-      default:
-        return <i>Invalid metric type: {type}</i>;
+    &.metric-histogram > dd {
+      margin-left: 145px;
     }
-  },
+  }
+`;
+
+class MetricDetails extends React.Component {
+ static propTypes = {
+   metrics: PropTypes.object,
+   metric: PropTypes.object.isRequired,
+   nodeId: PropTypes.string.isRequired,
+ };
+
+ static defaultProps = {
+   metrics: undefined,
+ }
+
+ componentDidMount() {
+   const { nodeId, metric: { full_name: fullName } } = this.props;
+   MetricsActions.add(nodeId, fullName);
+ }
+
+ componentWillUnmount() {
+   const { nodeId, metric: { full_name: fullName } } = this.props;
+   MetricsActions.remove(nodeId, fullName);
+ }
 
-  render() {
-    const metricName = this.props.metric.full_name;
-    const { nodeId } = this.props;
-    const metric = this.state.metrics && this.state.metrics[nodeId] && this.state.metrics[nodeId][metricName]
-      ? this.state.metrics[nodeId][metricName] : this.props.metric;
-    const type = lodash.capitalize(metric.type);
-    const details = this._formatDetailsForType(type, metric);
+ _formatDetailsForType = (type, metric) => {
+   switch (type) {
+     case 'Counter':
+       return <CounterDetails metric={metric} />;
+     case 'Gauge':
+       return <GaugeDetails metric={metric} />;
+     case 'Histogram':
+       return <HistogramDetails metric={metric} />;
+     case 'Meter':
+       return <MeterDetails metric={metric} />;
+     case 'Timer':
+       return <TimerDetails metric={metric} />;
+     default:
+       return <i>Invalid metric type: {type}</i>;
+   }
+ }
 
-    return (
-      <div className="metric">
-        <h3>{type}</h3>
+ render() {
+   const { nodeId, metric, metric: { full_name: metricName }, metrics } = this.props;
+   const currentMetric = metrics && metrics[nodeId] && metrics[nodeId][metricName]
+     ? metrics[nodeId][metricName] : metric;
```

Comment:
I know this was there when you got here, but do you think there is a clearer way to write this conditional?

Maybe this would work?

```suggestion
   const currentMetric = metrics?.[nodeId]?.[metricName] ?? metric;
```

Reply:
This works ! Thanks